### ISSUE TITLE
[Scala 2.12] scala.reflect.internal.Types$TypeError: illegal cyclic inheritance involving <refinement>

### ISSUE THREAD

**Issue by cchantep** [sbt/sbt#5424]
## steps

```
cd /tmp
cat > test.sh << EOF
set -e

git clone https://github.com/ReactiveMongo/ReactiveMongo-Streaming.git
cd ReactiveMongo-Streaming
git checkout b6e6c96
sbt ++2.12.10 error akka-stream/test:compile
EOF

docker run --rm -v $PWD/test.sh:/tmp/test.sh mozilla/sbt bash /tmp/test.sh
```

## problem

```
[error] ## Exception when compiling 3 sources to /ReactiveMongo-Streaming/akka-stream/target/scala-2.12/test-classes
[error] scala.reflect.internal.Types$TypeError: illegal cyclic inheritance involving <refinement>
```

Seems related to https://github.com/ReactiveMongo/ReactiveMongo-Streaming/blob/b6e6c96871ac43b7227e3641fe467c6511f262db/akka-stream/src/test/scala/GridFSSpec.scala

## expectation

No compilation error (without the current workaround)

**Comment by eed3si9n** [comment]
See https://github.com/scala/bug/issues/11880#issuecomment-583682673.

**Comment by cchantep** [comment]
Hi,

Sorry, but if I were able to provide a simpler reproducer, I would have.
And if I were able to figure about it has the same cause as the existing tickets I have read, I would have add elements there.

So if this bug is neither to be investigated at Scala compiler level or sbt one, I guess it won't be fixed for now.

I understand that ticket/issue triage is a challenge. I'm a little bit disappointed by such feedbacks.

Anyway thanks for you great work.
